### PR TITLE
 Consistently use axs to refer to a set of Axes

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

<s>Builds on top of #14079. You may wait with review until the above is in and I've rebased so that the changes are a bit smaller.</s> Is merged.

The intention of this PR is described in the addition to `pyplot.subplots()` docstring:

~~~
Typical idioms for handling the return value are::

    # using the variable ax for single a Axes
    fig, ax = plt.subplots()

    # using the variable axs for multiple Axes
    fig, axs = plt.subplots(2, 2)

    # using tuple unpacking for multiple Axes
    fig, (ax1, ax2) = plt.subplot(1, 2)
    fig, ((ax1, ax2), (ax3, ax4)) = plt.subplot(2, 2)

The names ``ax`` and pluralized ``axs`` are preferred over ``axes``
because for the latter it's not clear if it refers to a single
`~.axes.Axes` instance or a collection of these.
~~~

Everything else in the commit is only applying this to examples and tests.

### CODE REVIEW

File: examples/statistics/boxplot_vs_violin.py [hunk 1]
```diff
@@ -33,23 +33,23 @@
 all_data = [np.random.normal(0, std, 100) for std in range(6, 10)]
 
 # plot violin plot
-axes[0].violinplot(all_data,
+axs[0].violinplot(all_data,
```
Comment (anntzer) [inline examples/statistics/boxplot_vs_violin.py]: alignment

File: lib/matplotlib/tests/test_arrow_patches.py [hunk 2]
```diff
@@ -17,13 +17,13 @@ def test_fancyarrow():
     r = [0.4, 0.3, 0.2, 0.1, 0]
     t = ["fancy", "simple", mpatches.ArrowStyle.Fancy()]
 
-    fig, axes = plt.subplots(len(t), len(r), squeeze=False,
+    fig, axs = plt.subplots(len(t), len(r), squeeze=False,
```
Comment (anntzer) [inline lib/matplotlib/tests/test_arrow_patches.py]: alignment

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -754,13 +754,13 @@ def test_polar_theta_limits():
     theta_maxs = np.arange(50.0, 361.0, 90.0)
     DIRECTIONS = ('out', 'in', 'inout')
 
-    fig, axes = plt.subplots(len(theta_mins), len(theta_maxs),
+    fig, axs = plt.subplots(len(theta_mins), len(theta_maxs),
```
Comment (anntzer) [inline lib/matplotlib/tests/test_axes.py]: alignment

File: examples/specialty_plots/radar_chart.py [hunk 4]
```diff
@@ -162,13 +162,13 @@ def example_data():
     data = example_data()
     spoke_labels = data.pop(0)
 
-    fig, axes = plt.subplots(figsize=(9, 9), nrows=2, ncols=2,
+    fig, axs = plt.subplots(figsize=(9, 9), nrows=2, ncols=2,
```
Comment (anntzer) [inline examples/specialty_plots/radar_chart.py]: alignment

File: examples/statistics/violinplot.py [hunk 5]
```diff
@@ -28,59 +28,60 @@
...
 
-axes[0, 2].violinplot(data, pos, points=60, widths=0.7, showmeans=True,
-                      showextrema=True, showmedians=True, bw_method=0.5)
-axes[0, 2].set_title('Custom violinplot 3', fontsize=fs)
+axs[0, 2].violinplot(data, pos, points=60, widths=0.7, showmeans=True,
+                     showextrema=True, showmedians=True, bw_method=0.5)
+axs[0, 2].set_title('Custom violinplot 3', fontsize=fs)
 
-axes[0, 3].violinplot(data, pos, points=60, widths=0.7, showmeans=True,
+axs[0, 3].violinplot(data, pos, points=60, widths=0.7, showmeans=True,
```
Comment (anntzer) [inline examples/statistics/violinplot.py]: alignment (a bunch of them below, too)

Comment (timhoffm) [reply examples/statistics/violinplot.py]: Ha, got broken again due to rebasing :sad:. Thanks, fixed.

File: examples/statistics/boxplot.py [hunk 6]
```diff
@@ -67,28 +67,28 @@
...
+axs[0, 1].boxplot(data, flierprops=flierprops, medianprops=medianprops)
+axs[0, 1].set_title('Custom medianprops\nand flierprops', fontsize=fs)
 
-axes[0, 2].boxplot(data, whis='range')
-axes[0, 2].set_title('whis="range"', fontsize=fs)
+axs[0, 2].boxplot(data, whis='range')
+axs[0, 2].set_title('whis="range"', fontsize=fs)
 
-axes[1, 0].boxplot(data, meanprops=meanpointprops, meanline=False,
+axs[1, 0].boxplot(data, meanprops=meanpointprops, meanline=False,
```
Comment (anntzer) [inline examples/statistics/boxplot.py]: more alignment... (and below as well)
perhaps you can temporarily remove the flake8 rule excluding the check for these bad alignments to find all these cases locally :p

Comment (timhoffm) [reply examples/statistics/boxplot.py]: Done. Of course that was the last missing indent - well there are still lots, but not induced by this PR.

**Pull request merged**